inhibit/metric.go

```diff
@@ -0,0 +1,124 @@
+// Copyright The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package inhibit
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// InhibitorMetrics represents metrics associated to an inhibitor.
+type InhibitorMetrics struct {
+	// Inhibitor metrics
+	sourceAlertsCacheSize prometheus.Gauge
+	sourceAlertsIndexSize prometheus.Gauge
+	mutesDuration         *prometheus.SummaryVec
+
+	// Rule metrics
+	ruleSourceAlertsCacheSize *prometheus.GaugeVec
+	ruleSourceAlertsIndexSize *prometheus.GaugeVec
+	ruleMatchesDuration       *prometheus.SummaryVec
+	ruleMutesDuration         *prometheus.SummaryVec
+}
+
+// NewInhibitorMetrics returns a new InhibitorMetrics.
+func NewInhibitorMetrics(reg prometheus.Registerer) *InhibitorMetrics {
+	metrics := &InhibitorMetrics{
+		sourceAlertsCacheSize: prometheus.NewGauge(
+			prometheus.GaugeOpts{
+				Name: "alertmanager_inhibitor_source_alerts_cache_size",
```

Comment:
Size is a bit vague. Is this the number of entries in the cache?

```suggestion
				Name: "alertmanager_inhibitor_source_alerts_cache_entries",
```